New plot type "diurnal cycle" for monitoring diagnostics

axel-lauer opened the pull request:
## Description

This PR adds the plot type `diurnal_cycle` to the monitor diagnostics `monitor.py` and `multi_datasets.py`. The implementation is based on the code for the existing plot type `annual_cycle`.

An example has been added to the model evaluation recipe `recipe_model_evaluation_clouds_cycles.yml` and to the documentation of the monitor and the model evaluation recipes.

The new plot type requires the new preprocessor `local_solar_time`.

Closes #3498 

## Checklist

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

<!-- The next two lines turn the 🛠 and 🧪 below into hyperlinks -->
[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🛠][1] This pull request has a [descriptive title](https://docs.esmvaltool.org/en/latest/community/code_documentation.html#pull-request-title)
- [x] [🛠][1] Code is written according to the [code quality guidelines](https://docs.esmvaltool.org/en/latest/community/code_documentation.html#code-quality)
- [x] [🛠][1] [Documentation](https://docs.esmvaltool.org/en/latest/community/code_documentation.html#documentation) is available
- [ ] [🛠][1] [Tests](https://docs.esmvaltool.org/en/latest/community/code_documentation.html#tests) run successfully
- [x] [🛠][1] The [list of authors](https://docs.esmvaltool.org/en/latest/community/code_documentation.html#list-of-authors) is up to date
- [x] [🛠][1] Any changed dependencies have been [added or removed correctly](https://docs.esmvaltool.org/en/latest/community/code_documentation.html#dependencies)
- [ ] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/en/latest/community/code_documentation.html#pull-request-checks) were successful

### [New or updated recipe/diagnostic](https://docs.esmvaltool.org/en/latest/community/diagnostic.html)

- [x] [🧪][2] [Recipe runs successfully](https://docs.esmvaltool.org/en/latest/community/diagnostic.html#testing-recipes)
- [x] [🧪][2] [Recipe is well documented](https://docs.esmvaltool.org/en/latest/community/diagnostic.html#recipe-and-diagnostic-documentation)
- [x] [🧪][2] [Figure(s) and data](https://docs.esmvaltool.org/en/latest/community/diagnostic.html#diagnostic-output) look as expected from literature
- [x] [🛠][1] [Provenance information](https://docs.esmvaltool.org/en/latest/community/diagnostic.html#recording-provenance) has been added

LisaBock reviewed (changes requested):
Thanks @axel-lauer ! Looks mostly fine.

Some small comments:
- As I had difficulties to run the recipe `recipe_model_evaluation_clouds_cycles.yml` with one year of hourly ERA5 data I suggest to limit the time period to one month for the daily cycle.
- You added a figure to the documentation of the `recipe_monitor.yml`. Could you maybe also add it as one diagnostic block in the recipe?
- There is one print-statement in the code which should be removed.
- The line in the diurnal cycle of CERES-EBAF could be set "black" as other references.

axel-lauer commented:
Thanks @LisaBock for the review. I think I addressed all of your comments:

* Changed example `timerange` in `recipe recipe_model_evaluation_clouds_cycles.yml` to one month to speed up run time
* Added examples for calculating the diurnal cycle to the two monitoring recipes `recipe_monitor.yml` and `recipe_monitor_with_refs.yml`
* Added an example plot w/o reference dataset to the documentation of the monitoring recipe
* Removed print-statement in `multi_datasets.py`
* Added entry for line color (black) of CERES-EBAF

axel-lauer commented on esmvaltool/recipes/model_evaluation/recipe_model_evaluation_clouds_cycles.yml:
```diff
@@ -176,3 +234,42 @@ diagnostics:
                 color: black
```
Nice catch! Done.

LisaBock reviewed (approved):
Thanks @axel-lauer ! Looks all fine now to me and works perfectly!

axel-lauer commented:
I merged this into https://github.com/ESMValGroup/ESMValTool/tree/benchmarking_maps4monitoring as it is needed for the new benchmarking recipes contained in this branch. Merging conflicts are solved in https://github.com/ESMValGroup/ESMValTool/tree/benchmarking_maps4monitoring making it easier to merge the benchmarking recipes. This PR is now obsolete.
